### PR TITLE
Dropdown: Fix `testing-library/no-node-access` in tests

### PR DESCRIPTION
**Pull request by walbo**
## What?
With the recent work to improve the quality of tests, we fixed a bunch of ESLint rule violations. This PR fixes a few [`no-node-access`](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-node-access.md) rule violations in the `Dropdown` component. 

## Why?
The end goal is to enable that ESLint rule once all violations have been fixed.

## How?
Replace all `querySelector` and `querySelectorAll` with `*ByRole` and `*ByText`.

## Testing Instructions
Verify all tests still passes by running `npm run test:unit -- packages/components/src/dropdown/`

### CODE REVIEW

File: packages/components/src/dropdown/test/index.js
Comment (tyxla) [inline packages/components/src/dropdown/test/index.js]: Just like https://github.com/WordPress/gutenberg/pull/45694#discussion_r1018562140, we should probably be using `toBeVisible()`.

```suggestion
		expect( button ).toBeVisible();
```

File: packages/components/src/dropdown/test/index.js
Comment (tyxla) [inline packages/components/src/dropdown/test/index.js]:
```suggestion
		expect( screen.getByText( 'Content' ) ).toBeVisible();
```

File: packages/components/src/dropdown/test/index.js
Comment (tyxla) [inline packages/components/src/dropdown/test/index.js]:
```suggestion
		expect( screen.getByText( 'Content' ) ).toBeVisible();
```

File: packages/components/src/dropdown/test/index.js
Comment (tyxla) [inline packages/components/src/dropdown/test/index.js]: If we add `expanded: false` to the query, we won't need the extra `expect( button ).toHaveAttribute( 'aria-expanded', 'false' );` below. We could make a new query for the next assertion where we test for `[aria-expanded=true]`. No big deal, just an optional suggestion you could consider.

```suggestion
		const button = screen.getByRole( 'button', { expanded: false } );
```

**Review by tyxla — APPROVED** [review]
Another one bites the dust! Thanks, @walbo! 🚀

File: packages/components/src/dropdown/test/index.js
Comment (walbo) [reply packages/components/src/dropdown/test/index.js]: Smart. Updated in 2d62d9b4f4006b09200779ce7333f3bf18503e6b

File: packages/components/src/dropdown/test/index.js
Comment (walbo) [reply packages/components/src/dropdown/test/index.js]: Thanks. Needed to wrap them in `waitFor` because of the animation in `Popover`. See 87f8f79db31cf9036cc090b3f0634795de4ef746

File: packages/components/src/dropdown/test/index.js
Comment (tyxla) [inline packages/components/src/dropdown/test/index.js]: We could alternatively make a new query for the button with `expanded: true` if we don't want to get into the internals like this.

**Review by tyxla — APPROVED** [review]
🚀

File: packages/components/src/dropdown/test/index.js
Comment (walbo) [reply packages/components/src/dropdown/test/index.js]: Ah. Yes, my mistake. I'll update.

Comment (walbo) [reply packages/components/src/dropdown/test/index.js]: hmm. Does it make sense to create a new query when not using it in an `expect`?

or something like this:
```js
expect(
	screen.getByRole( 'button', { expanded: true } )
).toBeVisible();
```

Comment (tyxla) [reply packages/components/src/dropdown/test/index.js]: Yup something like that was what I had in mind 👍

Comment (walbo) [reply packages/components/src/dropdown/test/index.js]: Thanks for the clarification! Updated

**Review by Mamaduka — COMMENTED** [review]
I think we can merge this once merge conflicts are resolved.

**Comment by tyxla** [comment]
Looks like this has already been addressed in #45911 and this one can be closed.

**Pull request closed without merging**
